server/src/main/java/org/elasticsearch/bootstrap/ElasticsearchUncaughtExceptionHandler.java

```diff
@@ -54,21 +53,11 @@ static boolean isFatalUncaught(Throwable e) {
     void onFatalUncaught(final String threadName, final Throwable t) {
         final String message = "fatal error in thread [" + threadName + "], exiting";
         logger.error(message, t);
-        Terminal.DEFAULT.errorPrintln(message);
-        t.printStackTrace(Terminal.DEFAULT.getErrorWriter());
-        // Without a final flush, the stacktrace may not be shown before ES exits
-        Terminal.DEFAULT.flush();
```

Comment:
If we are still using the terminal to print, do we need to do this flush before we exit?

Reply:
I thought no, but there is a small edge case where logging has not been setup in which we might not flush (otherwise we flush indirectly in printLogsSuggestion). I've added an explicit flush right before we exit.